src/Sulu/Bundle/AdminBundle/Resources/js/components/ImageSectorSelection/ImageSectorSelection.js

```diff
@@ -0,0 +1,130 @@
+        initialSelection?: SelectionData,
+        minWidth?: number,
+        minHeight?: number,
+        onChange?: (s: SelectionData) => void,
+        src: string,
+    };
+
+    image: Image;
+    rounding = new RoundingNormalizer();
+
+    @observable imageLoaded = false;
+    @observable containerWidth: number;
+    @observable containerHeight: number;
+
+    naturalHorizontalToReal = (h: number) => h * this.imageResizedWidth / this.image.naturalWidth;
+    realHorizontalToNatural = (h: number) => h * this.image.naturalWidth / this.imageResizedWidth;
+    naturalVerticalToReal = (v: number) => v * this.imageResizedHeight / this.image.naturalHeight;
+    realVerticalToNatural = (v: number) => v * this.image.naturalHeight / this.imageResizedHeight;
+
+    naturalDataToReal(data: SelectionData): SelectionData {
+        return {
+            width: this.naturalHorizontalToReal(data.width),
+            height: this.naturalVerticalToReal(data.height),
+            left: this.naturalHorizontalToReal(data.left),
+            top: this.naturalVerticalToReal(data.top),
+        };
+    }
+
+    realDataToNatural(data: SelectionData): SelectionData {
+        return {
+            width: this.realHorizontalToNatural(data.width),
+            height: this.realVerticalToNatural(data.height),
+            left: this.realHorizontalToNatural(data.left),
+            top: this.realVerticalToNatural(data.top),
+        };
+    }
+
+    @action componentWillMount() {
+        this.image = new Image();
+        this.image.onload = action(() => this.imageLoaded = true);
+        this.image.onerror = () => log.error('Failed to preload image ' + this.props.src);
+        this.image.src = this.props.src;
+    }
+
+    @computed get imageResizedHeight(): number {
+        if (this.imageTouchesHoriziontalBorders()) {
+            return Math.min(this.image.naturalHeight, this.containerHeight);
+        } else {
+            return this.imageResizedWidth * this.image.naturalHeight / this.image.naturalWidth;
+        }
+    }
+
+    @computed get imageResizedWidth(): number {
+        if (this.imageTouchesHoriziontalBorders()) {
+            return this.imageResizedHeight * this.image.naturalWidth / this.image.naturalHeight;
+        } else {
+            return Math.min(this.image.naturalWidth, this.containerWidth);
+        }
+    }
+
+    imageTouchesHoriziontalBorders() {
+        const imageHeightToWidth = this.image.naturalHeight / this.image.naturalWidth;
+        const containerHeightToWidth = this.containerHeight / this.containerWidth;
+        return imageHeightToWidth > containerHeightToWidth;
+    }
+
+    handleRectangleSelectionChange = (data: SelectionData) => {
+        const computedData = this.rounding.normalize(this.realDataToNatural(data));
+        if (this.props.onChange) {
+            this.props.onChange(computedData);
+        }
+    };
+
+    readContainerDimensions = (container: HTMLElement) => {
```

Comment:
I thought about it a little bit more, and I think that's something we could extract into a [higher order component](https://facebook.github.io/react/docs/higher-order-components.html). You can also check the [`withToolbar`](https://github.com/sulu/sulu/blob/develop/src/Sulu/Bundle/AdminBundle/Resources/js/containers/Toolbar/withToolbar.js) implementation.